data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/oidc/TokenProvider.java

```diff
@@ -15,30 +15,51 @@
  */
 package dev.knative.eventing.kafka.broker.core.oidc;
 
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
 import dev.knative.eventing.kafka.broker.core.NamespacedName;
 import io.fabric8.kubernetes.api.model.authentication.TokenRequest;
 import io.fabric8.kubernetes.api.model.authentication.TokenRequestBuilder;
-import io.fabric8.kubernetes.client.Config;
-import io.fabric8.kubernetes.client.ConfigBuilder;
 import io.fabric8.kubernetes.client.KubernetesClient;
-import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+
+import java.util.concurrent.TimeUnit;
 
 public class TokenProvider {
 
     private final KubernetesClient kubernetesClient;
+    private final Cache<String, String> tokenCache;
+
+    public TokenProvider(KubernetesClient kubernetesClient) {
+        this.kubernetesClient = kubernetesClient;
```

Comment:
why this change? We can build the kubernetesClient without anything else and does not require the caller to setup one if no client is arround.
But we could overload it :shrug: